libs/langgraph/langgraph/prebuilt/interrupt.py

```diff
@@ -0,0 +1,74 @@
+from typing import (
+    Literal,
+    Optional,
+    Union,
+)
+
+from typing_extensions import TypedDict
+
+
+class HumanInterruptConfig(TypedDict):
+    """Configuration that defines what actions are allowed for a human interrupt.
+
+    This controls the available interaction options when the graph is paused for human input.
+
+    Attributes:
+        allow_ignore (bool): Whether the human can choose to ignore/skip the current step
+        allow_respond (bool): Whether the human can provide a text response/feedback
+        allow_edit (bool): Whether the human can edit the provided content/state
+        allow_accept (bool): Whether the human can accept/approve the current state
+    """
+
+    allow_ignore: bool
+    allow_respond: bool
+    allow_edit: bool
+    allow_accept: bool
+
+
+class ActionRequest(TypedDict):
+    """Represents a request for human action within the graph execution.
+
+    Contains the action type and any associated arguments needed for the action.
+
+    Attributes:
+        action (str): The type or name of action being requested (e.g., "Approve XYZ action")
+        args (dict): Key-value pairs of arguments needed for the action
+    """
+
+    action: str
+    args: dict
+
+
+class HumanInterrupt(TypedDict):
```

Comment:
Consider adding an inline example of a human interrupt and then a corresponding example for human response